pyccel/codegen/printing/ccode.py

```diff
@@ -696,11 +696,17 @@ def _print_IndexedElement(self, expr):
                         inds[i] = Slice(ind, PyccelAdd(ind, LiteralInteger(1)), LiteralInteger(1))
                 inds = [self._print(i) for i in inds]
                 return "array_slicing(%s, %s, %s)" % (base_name, expr.rank, ", ".join(inds))
-            inds = [self._print(i) for i in inds]
+            inds = [self._cast_to(i, NativeInteger(), 8) for i in inds]
         else:
             raise NotImplementedError(expr)
         return "%s.%s[get_index(%s, %s)]" % (base_name, dtype, base_name, ", ".join(inds))
 
+    def _cast_to(self, expr, dtype, precision):
+        if (expr.dtype == dtype and expr.precision != precision):
```

Comment:
Shouldn't this be:
`(expr.dtype != dtype or expr.precision != precision)`
?

Reply:
i wanted the function to be unable to cast to a different type. (float -> int)

Reply:
That makes sense for indexing arguments, but I'm not sure it makes sense for a function called `_cast_to`.

I am thinking about cases where I might re-use this later. If I did `cast_to(obj, Real, 4)` where `obj` is an integer, I would be very surprised if I got back `expr`.

Maybe the function should be renamed? `_choose_precision`? `_enforce_precision`?
Or maybe we should check that the argument is an integer elsewhere and then leave this as a standard casting function?

What do you think?

Reply:
if we can unsure the arguments to be integers we can leave it as a standard casting function.

Reply:
I tested and it didn't raise an error :/ so it probably should be added as a warning or an error in the semantic stage. We usually say that pyccel only works on valid python code though

Reply:
yeah it doesn't raise the error , should all `indexedElement` raise an error if the arguments are not integers?

Reply:
I think that would be a good idea. In theory numpy arrays can be indexed with booleans too (see issue #238) so it is important that we raise an error if the index is something that we don't handle